site/learn/expert-advice/dev-tools/uuid-generator.html

```diff
@@ -30,4 +31,27 @@
   </form>
 </div>
 
-<script type="text/javascript" src="/assets/js/UUIDGenerator-0.1.0.js" defer></script>
\ No newline at end of file
+<h3>UUID Frequently Asked Questions</h3>
+<h4>What is a UUID?</h4>
+<p>A UUID (Universally Unique Identifier) is a string of characters that is used to uniquely identify an object. You may also see the term GUID (globally unique identifier) which is also used. Both UUIDs and GUIDs are generated using random numbers, which makes it extremely unlikely that two of them will be the same.
+</p>
+<br>
+<h4>What version UUID generator is this tool?</h4>
+<p>This UUID generator generates a version 4 universally unique identifier. A version 4 UUID is made up of 32 hexadecimal digits (128 bits), divided into five groups separated by hyphens in this format: Xxxxxxxx-xxxx-4xxx-xxxx-xxxxxxxxxxxx.
+</p>
+<p>
+The main advantage of version 4 UUIDs is that they are generated using random numbers, which makes it extremely unlikely that two UUIDs will be the same. The sheer number of possible outcome combinations (2^128) would make it nearly impossible to generate a duplicate for practical purposes.
```

Comment:
```suggestion
The main advantage of version 4 UUIDs is that they are generated using random numbers, which makes it extremely unlikely that two UUIDs will be the same. The sheer number of possible outcome combinations (2^128) makes duplicates extremely unlikely.
```

Reply:
Suggestion: 

> The sheer number of possible outcome combinations (2^128) makes the possibility of a duplicate value being generated so unlikely that in practice we can ignore the possibility. 

If we want to be completely accurate, (and I think we should since nerds will read this) this is actually 2^122 because 4 bits are used to indicate the version of the UUID, and 2 bits to indicate the variant,, so we start with 128 - 6, leaves 122. 

This means there are 2^122 variations (or 5.3×103, or 5.3 undecillion)